src/Stratis.FederatedPeg.Features.FederationGateway/TemplateTransactionStore.cs

```diff
@@ -0,0 +1,216 @@
+﻿using System;
+using System.IO;
+using System.Threading.Tasks;
+using DBreeze;
+using DBreeze.DataTypes;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Configuration;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.FederatedPeg.Features.FederationGateway
+{
+    /// <summary></summary>
+    public interface ITemplateTransactionStore : IDisposable
+    {
+        /// <summary>
+        /// Persist the template transaction into the database.
+        /// </summary>
+        /// <param name="templateTransaction">Template transaction to be inserted.</param>
+        Task PutAsync(TemplateTransaction templateTransaction);
+
+        /// <summary>
+        /// Get the template transaction from the database, identified by a hash.
+        /// </summary>
+        /// <param name="hash">A template transaction hash.</param>
+        /// <returns>The template transaction identified by the hash (or null if not found).</returns>
+        Task<TemplateTransaction> GetAsync(uint256 hash);
+
+        /// <summary>
+        /// Delete a template transaction from the database, identified by a hash.
+        /// </summary>
+        /// <param name="hash">Hash of the template transaction to be deleted.</param>
+        Task DeleteAsync(uint256 hash);
+
+        /// <summary>
+        /// Determine if a template transaction already exists in the database.
+        /// </summary>
+        /// <param name="hash">The hash of the template transaction.</param>
+        /// <returns><c>true</c> if the template transaction can be found in the database, otherwise return <c>false</c>.</returns>
+        Task<bool> ExistAsync(uint256 hash);
+    }
+
+    public class TemplateTransactionStore : ITemplateTransactionStore
```

Comment:
If I understand that correctly, this is the store where we plan to have all the different crosschain transfers and their different statuses ?